core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -186,24 +224,12 @@ UNWIND(	.cantunwind)
 
 	bootargs_entry
 
-	/* Enable alignment checks and disable data and instruction cache. */
-	read_sctlr r0
-#if defined(CFG_SCTLR_ALIGNMENT_CHECK)
-	orr	r0, r0, #SCTLR_A
-#else
-	bic	r0, r0, #SCTLR_A
-#endif
-	bic	r0, r0, #SCTLR_C
-	bic	r0, r0, #SCTLR_I
-#if defined(CFG_HWSUPP_MEM_PERM_WXN) && defined(CFG_CORE_RWDATA_NOEXEC)
-	orr	r0, r0, #(SCTLR_WXN | SCTLR_UWXN)
-#endif
-	write_sctlr r0
-	isb
-
 	/* Early ARM secure MP specific configuration */
 	bl	plat_cpu_reset_early
 
+	set_sctlr
+	isb
```

Comment:
True but i preferred this way as it is aligned by the generic_entry_a64.S and make it clear there is a synchro.